custom_components/thermal_comfort/sensor.py

```diff
@@ -648,11 +649,11 @@ async def temperature_state_listener(self, event):
     async def _new_temperature_state(self, state):
         if _is_valid_state(state):
             unit = state.attributes.get(ATTR_UNIT_OF_MEASUREMENT)
-            temp = util.convert(state.state, float)
+            temp = convert(state.state, float)
             self.extra_state_attributes[ATTR_TEMPERATURE] = temp
             # convert to celsius if necessary
             if unit == TEMP_FAHRENHEIT:
```

Comment:
This `if` may be skipped now (may be while future refactoring stage):
```python
    @classmethod
    def convert(
        cls, value: float, from_unit: str, to_unit: str, *, interval: bool = False
    ) -> float:
        """Convert a temperature from one unit to another."""
        cls._check_arguments(value, from_unit, to_unit)

        if from_unit == to_unit:
            return value
...
```

Reply:
Not sure what you mean here

Reply:
We may not check that `unit` in `thermal_comfort/sensor.py` is `TEMP_FAHRENHEIT`, before running conversion to Celsius, because converter will check if `from_unit` and `to_unit` is differ and will do real conversion only if needed (as shown in quoted code).
So we may use just something like
```python
temp = TemperatureConverter.convert(temp, unit, TEMP_CELSIUS)
```
Also it solve potential problem if someone use temperature in Kelvin (just for fun).

Just little code optimization.

Reply:
Good idea, thank you. We still need to check if unit is None though.

Reply:
> Good idea, thank you. We still need to check if unit is `None` though.

Then `HomeAssistantError` will be raised.
Or we may use something like
```python
unit = state.attributes.get(ATTR_UNIT_OF_MEASUREMENT, MY_BEST_GUESS_UNIT)
```
to avoid `None` unit.